scripts/base/protocols/rdp/main.zeek

```diff
@@ -187,7 +187,22 @@ event rdp_client_core_data(c: connection, data: RDP::ClientCoreData) &priority=5
 	{
 	set_session(c);
 
-	c$rdp$keyboard_layout       = RDP::languages[data$keyboard_layout];
+	if (data$keyboard_layout in RDP::languages)
+		{
+		c$rdp$keyboard_layout = RDP::languages[data$keyboard_layout];
+		}
+	else
+		{
+		if (data$keyboard_layout & 0xffff in RDP::languages)
+			{
+			c$rdp$keyboard_layout = fmt("%s (Best Guess)", RDP::languages[data$keyboard_layout & 0xffff]);
```

Comment:
Should we include the original value E.g. `"(Best Guess of %d)"` or `%08x`? We otherwise don't have the actual value in the log anymore.

Reply:
In the old code, if there was a match, just the human readable value was listed and the decimal numerical if not. The change here expands the matching capability, but in the event of no match, it still logs in the previous form.

I agree with this though, if we aren't a 100% match, why not include the raw in the output? I'll make the change shortly.